og_uk_calibrate/get_micro_data.py

```diff
@@ -12,13 +12,68 @@
 from openfisca_uk import Microsimulation
 import pandas as pd
 import warnings
-
+from openfisca_uk.api import *
+from openfisca_data import FRS, SynthFRS
+
+if len(FRS.years) == 0:
+    print("Using synthetic dataset.")
+    SynthFRS.save(
+        "https://github.com/nikhilwoodruff/openfisca-uk-data/releases/download/synth-frs/synth_frs_2018.h5",
+        2018,
+    )
+    dataset = SynthFRS
+else:
+    dataset = FRS
 warnings.filterwarnings("ignore")
 
 CUR_PATH = os.path.split(os.path.abspath(__file__))[0]
 DATA_LAST_YEAR = 2021  # this is the last year data are extrapolated for
 
 
+def get_mtrs_employment_income(reform, **kwargs):
+    baseline = Microsimulation(reform, dataset=dataset, **kwargs)
```

Comment:
@nikhilwoodruff  I'm trying to understand more about how to interact with OpenFisca-UK:  why aren't you using the `deriv` method here to compute MTRs?  A `Microsimulation` object had already been created and here you create two more just to compute one MTR.

Reply:
@jdebacker The reason I didn't use `sim.deriv` was mainly to do with the fact that the MTRs had to be household-level, rather than individual level (probably possible with manipulating the outputs of `sim.deriv`, but that would return person-level MTRs). But you're right about the redundancy, either way we need at least 3 microsimulations (baseline, baseline + employment bonus, baseline + savings bonus, and here we're using 4 (can take out baseline from the function).

Reply:
@nikhilwoodruff Thanks for this clarification.  The ["else if" on line 449 of `simulation.py`](https://github.com/PSLmodels/openfisca-uk/blob/ab5f893298b827e906ad07a8dffb815ea3d185be/openfisca_uk/tools/simulation.py#L449) seems to suggest that `Microsimulation.deriv` can allow for `target_entity` values that are not the same as the `wrt_entity` values.  Am I missing something?  But I see no way to specify the `target_entity` in the call to the `deriv` method (e.g., at the `calc` method seems to allow with the `map_to` argument).  

If these mappings between different entity types in different functions are documented somewhere, please let me know.

Reply:
@jdebacker that's right. At the moment the target entity and the wrt entity are fixed to the entities of the variables (e.g. if you provided "net_income" as the target variable, the target entity would be Person, but if you provided "household_net_income", it'd be Household). I'll add a manual override e.g. to specify net_income mapped to households.